beacon-chain/state/getters.go

```diff
@@ -717,56 +614,94 @@ func (b *BeaconState) PreviousJustifiedCheckpoint() *ethpb.Checkpoint {
 	if !b.HasInnerState() {
 		return nil
 	}
-	if b.state.PreviousJustifiedCheckpoint == nil {
-		return nil
-	}
-
-	b.lock.RLock()
-	defer b.lock.RUnlock()
-
-	return CopyCheckpoint(b.state.PreviousJustifiedCheckpoint)
+	return b.safeCopyCheckpoint(b.state.PreviousJustifiedCheckpoint)
 }
 
 // CurrentJustifiedCheckpoint denoting an epoch and block root.
 func (b *BeaconState) CurrentJustifiedCheckpoint() *ethpb.Checkpoint {
 	if !b.HasInnerState() {
 		return nil
 	}
-	if b.state.CurrentJustifiedCheckpoint == nil {
+	return b.safeCopyCheckpoint(b.state.CurrentJustifiedCheckpoint)
+}
+
+// FinalizedCheckpoint denoting an epoch and block root.
+func (b *BeaconState) FinalizedCheckpoint() *ethpb.Checkpoint {
+	if !b.HasInnerState() {
 		return nil
 	}
+	return b.safeCopyCheckpoint(b.state.FinalizedCheckpoint)
+}
 
+// FinalizedCheckpointEpoch returns the epoch value of the finalized checkpoint.
+func (b *BeaconState) FinalizedCheckpointEpoch() uint64 {
+	if !b.HasInnerState() {
+		return 0
+	}
+	if b.state.FinalizedCheckpoint == nil {
+		return 0
+	}
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return CopyCheckpoint(b.state.CurrentJustifiedCheckpoint)
+	return b.state.FinalizedCheckpoint.Epoch
 }
 
-// FinalizedCheckpoint denoting an epoch and block root.
-func (b *BeaconState) FinalizedCheckpoint() *ethpb.Checkpoint {
-	if !b.HasInnerState() {
+func (b *BeaconState) safeCopy2DByteSlice(input [][]byte) [][]byte {
+	if input == nil {
```

Comment:
perform a state check here first

Reply:
This function doesn't access any state, it just handles the input.